web/src/settings_muted_topics.js

```diff
@@ -3,52 +3,83 @@ import $ from "jquery";
 import render_user_topic_ui_row from "../templates/user_topic_ui_row.hbs";
 
 import * as ListWidget from "./list_widget";
+import {page_params} from "./page_params";
+import * as settings_config from "./settings_config";
 import * as ui from "./ui";
 import * as user_topics from "./user_topics";
 
 export let loaded = false;
 
 export function populate_list() {
-    const all_muted_topics = user_topics.get_user_topics_for_visibility_policy(
-        user_topics.all_visibility_policies.MUTED,
-    );
-    const $muted_topics_table = $("#muted_topics_table");
-    const $search_input = $("#muted_topics_search");
-
-    ListWidget.create($muted_topics_table, all_muted_topics, {
-        name: "muted-topics-list",
+    const all_user_topics = [];
+    const visibility_policies = Object.values(user_topics.all_visibility_policies);
+    for (const visibility_policy of visibility_policies) {
+        const user_topics_list =
+            user_topics.get_user_topics_for_visibility_policy(visibility_policy);
+        for (const user_topic of user_topics_list) {
+            user_topic.visibility_policy = visibility_policy;
+            all_user_topics.push(user_topic);
+        }
+    }
```

Comment:
How does this loop work with `Inherit` -- does it just not end up finding any? Might be worth a comment.

Also, should we tweak `get_user_topics_for_visibility_policy` to include the policy value in the dictionaries it returns to avoid the mutation in this second loop? I don't think we use that function in many places elsewhere, so we might as well give it a format that this wants.

Reply:
> Also, should we tweak get_user_topics_for_visibility_policy to include the policy value in the dictionaries it returns to avoid the mutation in this second loop?

**Added a prep commit for this.**

> How does this loop work with Inherit -- does it just not end up finding any? Might be worth a comment.

**Added a comment.**

Actually, we only maintain the record of topics having visibility_policy other than `INHERIT`; `INHERIT` is used to remove topics from the record. -- [update_user_topics](https://github.com/zulip/zulip/blob/main/web/src/user_topics.js#L22) for reference.